src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -0,0 +1,59 @@
+package se.liu.ida.hefquin.engine.data.impl;
+
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.RDFDataMgr;
+
+import se.liu.ida.hefquin.engine.data.Triple;
+import se.liu.ida.hefquin.engine.data.VocabularyMapping;
+import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
+import se.liu.ida.hefquin.engine.query.TriplePattern;
+
+public class VocabularyMappingImpl implements VocabularyMapping{
+	
+	protected final Model vocabularyMapping;
+	
+	public VocabularyMappingImpl() {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+	}
+	
+	
+	// Source: https://jena.apache.org/documentation/io/rdf-input.html
+	public VocabularyMappingImpl(final String rdfFile) {
+		vocabularyMapping = RDFDataMgr.loadModel(rdfFile); //.nt file for N-Triple
+	}
+	
+	public VocabularyMappingImpl(final Set<Triple> tps) {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+		Iterator<Triple> i = tps.iterator();
+		while(i.hasNext()) {
+			Triple tp = i.next();
+			vocabularyMapping.asStatement(tp.asJenaTriple());
```

Comment:
Are you sure that this adds the statement into the model? The javadoc of the method `asStatement` is a bit cryptic, but it seems to me that it takes a Triple and transforms it into a Statement, but I'm not sure that it also adds the Statement object to the Model. Please, check this, if it doesn't work, then maybe you need to do something like:

```
vocabularyMapping.add(vocabularyMapping.asStatement(t.asJEnaTriple());
```

Reply:
Another even easier option: instead of creating a `Model` initially, create a Jena `Graph` object (there is a `GraphFactory` helper class to do that), add the triples into this `Graph`, and finally create the `Model` based on that `Graph`.

In fact, it may not be necessary to have a `Model` at all but, instead, simply have it only as a `Graph` object.

Reply:
If the graph isn't made into a Model in the constructor, should an extra function be added to create a model out of it, to be able to query the model later? Or is it okay to create a model out of it directly in the getMappings function?

Reply:
I think I missed that find can be used on the graph directly, and with that a Model would not be necessary right?

Reply:
Right.